devito/interfaces.py

```diff
@@ -716,10 +726,16 @@ def _indices(cls, **kwargs):
                       automatically infer dimension symbols.
         :return: Dimension indices used for each axis.
         """
-        save = kwargs.get('save', None)
-        tidx = time if save else t
-        _indices = DenseData._indices(**kwargs)
-        return tuple([tidx] + list(_indices))
+        # Get dimensions
+        dimensions = kwargs.get('dimensions', None)
+        # Add time index if not specified
+        if dimensions is None or (t not in dimensions and time not in dimensions):
+            save = kwargs.get('save', None)
+            tidx = time if save else t
+            _indices = DenseData._indices(**kwargs)
```

Comment:
does ``super(TimeData, self)._indices(**kwargs)`` work ?

Reply:
This was there before, shows as diff because of the if